xrmocap/core/evaluation/fourd_association_evaluation.py

```diff
@@ -0,0 +1,440 @@
+                continue
+            gt_kps3d_idxs = np.where(
+                np.sum(gt_kps3d_mask[frame_idx], axis=1) > 0)[0]
+            for gt_kps3d_idx in gt_kps3d_idxs:
+                f_gt_kps3d = gt_kps3d[frame_idx][gt_kps3d_idx]
+                f_pred_kps3d = pred_kps3d[frame_idx][
+                    np.sum(pred_kps3d_mask[frame_idx], axis=1) > 0]
+                if len(f_pred_kps3d) == 0:
+                    continue
+
+                dist = vectorize_distance(f_gt_kps3d[np.newaxis], f_pred_kps3d)
+                f_pred_kps3d = f_pred_kps3d[np.argmin(dist[0])]
+
+                if np.all((f_pred_kps3d == 0)):
+                    continue
+
+                # MPJPE
+                f_pred_keypoints = Keypoints(
+                    kps=np.concatenate(
+                        (f_pred_kps3d, np.ones_like(f_pred_kps3d[..., 0:1])),
+                        axis=-1),
+                    convention=pred_kps3d_convention)
+                f_gt_keypoints = Keypoints(
+                    kps=np.concatenate(
+                        (f_gt_kps3d, np.ones_like(f_gt_kps3d[..., 0:1])),
+                        axis=-1),
+                    convention=gt_kps3d_convention)
+                mpjpe.append(
+                    compute_mpjpe(
+                        f_pred_keypoints, f_gt_keypoints, align=True))
+
+                # PA-MPJPE
+                _, _, rotation, scaling, transl = compute_similarity_transform(
+                    f_gt_kps3d, f_pred_kps3d, compute_optimal_scale=True)
+                pred_kps3d_pa = (scaling * f_pred_kps3d.dot(rotation)) + transl
+
+                pred_keypoints_pa = Keypoints(
+                    kps=np.concatenate(
+                        (pred_kps3d_pa, np.ones_like(pred_kps3d_pa[..., 0:1])),
+                        axis=-1),
+                    convention=pred_kps3d_convention)
+                pa_mpjpe_i = compute_mpjpe(
+                    pred_keypoints_pa, f_gt_keypoints, align=True)
+                pa_mpjpe.append(pa_mpjpe_i)
+
+                pck_50.append(np.mean(pa_mpjpe_i <= (pck_50_thres / scale)))
+                pck_100.append(np.mean(pa_mpjpe_i <= (pck_100_thres / scale)))
+        mpjpe = np.asarray(mpjpe) * scale  # m to mm
+        pa_mpjpe = np.asarray(pa_mpjpe) * scale  # m to mm
+        mpjpe_mean, mpjpe_std = np.mean(mpjpe), np.std(mpjpe)
+        pa_mpjpe_mean, pa_mpjpe_std = np.mean(pa_mpjpe), np.std(pa_mpjpe)
+        pck_50 = np.mean(pck_50) * 100.  # percentage
+        pck_100 = np.mean(pck_100) * 100.  # percentage
+        self.logger.info(f'MPJPE: {mpjpe_mean:.2f} ± {mpjpe_std:.2f} mm')
+        self.logger.info(f'PA-MPJPE: {pa_mpjpe_mean:.2f} ±'
+                         f'{pa_mpjpe_std:.2f} mm')
+        self.logger.info(f'PCK@{pck_50_thres}mm: {pck_50:.2f} %')
+        self.logger.info(f'PCK@{pck_100_thres}mm: {pck_100:.2f} %')
+        return pck_50, pck_100, mpjpe, pa_mpjpe
+
+    def align_keypoints3d(self, pred_keypoints3d: Keypoints,
```

Comment:
It looks like MVPose. Shall we create a folder `alignment` at `xrmocap/core/evaluation` and put the common functions there?

Reply:
I will adopt it.